File: BraintreeCore/src/test/java/com/braintreepayments/api/AuthorizationUnitTest.java
package com.braintreepayments.api;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class AuthorizationUnitTest {

    @Test
    public void fromString_returnsValidClientTokenWhenBase64() {
        Authorization authorization = Authorization.fromString(Fixtures.BASE64_CLIENT_TOKEN);

        assertTrue(authorization instanceof ClientToken);
    }

    @Test
    public void fromString_returnsValidClientTokenWhenBase64IncludesSpaces() {
        Authorization authorization = Authorization.fromString(Fixtures.BASE64_CLIENT_TOKEN_WITH_SPACES);

        assertTrue(authorization instanceof ClientToken);
    }

    @Test
    public void fromString_returnsValidTokenizationKey() {
        Authorization authorization = Authorization.fromString(Fixtures.TOKENIZATION_KEY);

        assertTrue(authorization instanceof TokenizationKey);
    }

    @Test
    public void fromString_returnsValidTokenizationKeyIncludesSpaces() {
        Authorization authorization = Authorization.fromString(Fixtures.TOKENIZATION_KEY_WITH_SPACES);

        assertTrue(authorization instanceof TokenizationKey);
    }

    @Test
    public void fromString_whenPassedNull_returnsInvalidToken() {
        Authorization result = Authorization.fromString(null);

        assertTrue(result instanceof InvalidAuthorization);
    }

    @Test
    public void fromString_whenPassedAnEmptyString_returnsInvalidToken() {
        Authorization result = Authorization.fromString("");

        assertTrue(result instanceof InvalidAuthorization);
    }

    @Test
    public void fromString_whenPassedJunk_returnsInvalidToken() {
        Authorization result = Authorization.fromString("not authorization");

        assertTrue(result instanceof InvalidAuthorization);
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/AuthorizationUnitTest.kt
package com.braintreepayments.api

import com.braintreepayments.api.Authorization.Companion.fromString
import org.robolectric.RobolectricTestRunner
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class AuthorizationUnitTest {
    @Test
    fun fromString_returnsValidClientTokenWhenBase64() {
        val authorization = fromString(Fixtures.BASE64_CLIENT_TOKEN)
        assertTrue(authorization is ClientToken)
    }

    @Test
    fun fromString_returnsValidClientTokenWhenBase64IncludesSpaces() {
        val authorization = fromString(Fixtures.BASE64_CLIENT_TOKEN_WITH_SPACES)
        assertTrue(authorization is ClientToken)
    }

    @Test
    fun fromString_returnsValidTokenizationKey() {
        val authorization = fromString(Fixtures.TOKENIZATION_KEY)
        assertTrue(authorization is TokenizationKey)
    }

    @Test
    fun fromString_returnsValidTokenizationKeyIncludesSpaces() {
        val authorization = fromString(Fixtures.TOKENIZATION_KEY_WITH_SPACES)
        assertTrue(authorization is TokenizationKey)
    }

    @Test
    fun fromString_whenPassedNull_returnsInvalidToken() {
        val result = fromString(null)
        assertTrue(result is InvalidAuthorization)
    }

    @Test
    fun fromString_whenPassedAnEmptyString_returnsInvalidToken() {
        val result = fromString("")
        assertTrue(result is InvalidAuthorization)
    }

    @Test
    fun fromString_whenPassedJunk_returnsInvalidToken() {
        val result = fromString("not authorization")
        assertTrue(result is InvalidAuthorization)
    }
}
